### PR TITLE
Merge the pyi files into py files of optimizer

### PR DESCRIPTION
**Pull request by david20571015**
Merge the interfaces in pyi files into py files in `torch/optim`.

### CODE REVIEW

**Comment by david20571015** [comment]
Hello reviewers, I am working on merging all interfaces in the `torch/optim`. I have currently completed the merge for one interface (`torch/optim/adadelta.py`), and I would appreciate your feedback on my work. If you approve of my approach, I will continue merging the remaining interfaces. Thanks.

File: torch/optim/adadelta.py [hunk 1]
```diff
@@ -394,16 +395,10 @@ def _multi_tensor_adadelta(
             torch._foreach_add_(device_state_steps, 1)
 
         if maximize:
-            device_grads = torch._foreach_neg(device_grads)
+            torch._foreach_neg_(device_grads)
```
Comment (janeyx99) [inline torch/optim/adadelta.py:398]: this would inplace modify the grads which is not okay

**Review by janeyx99 — COMMENTED** [review]
thanks for taking this on, reviewed!

File: torch/optim/adadelta.py (hunk 1 above)
Comment (janeyx99) [reply torch/optim/adadelta.py]: @david20571015 let's just type ignore

Comment (david20571015) [reply torch/optim/adadelta.py]: ok

**Review by janeyx99 — APPROVED** [review]
thanks!!

**Pull request closed without merging**